Source/JavaScriptCore/runtime/ArrayConstructor.cpp

```diff
@@ -79,7 +79,7 @@ JSArray* constructArrayWithSizeQuirk(JSGlobalObject* globalObject, ArrayAllocati
     
     uint32_t n = length.toUInt32(globalObject);
     if (n != length.toNumber(globalObject)) {
-        throwException(globalObject, scope, createRangeError(globalObject, "Array size is not a small enough positive integer."_s));
+        throwException(globalObject, scope, createRangeError(globalObject, "Array length must be a positive integer of safe magnitude."_s));
```

Comment:
Can we share the error string in ArrayConstructor.h? Something like `constexpr ASCIILiteral arrayInvalidLengthError = "Array length must be a positive integer of safe magnitude."_s`

Reply:
Can do!